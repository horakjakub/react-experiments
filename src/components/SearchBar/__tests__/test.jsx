import React, {ChangeEvent} from 'react';
import {render, fireEvent, screen, createEvent} from '@testing-library/react';
import {Search, withOnChange} from '../index';

describe('<Search />', () => {
  it('renders correctly', () => {
    const {container} = render(<Search phrase="" setPhrase={() => {}} />);
    expect(container).toMatchSnapshot();
  });

  it('should pass typed phrase into "setPhrase" prop function', () => {
    const spy = jest.fn();
    const {getByPlaceholderText} = render(<Search phrase="" setPhrase={spy} />);
    const searchEl = getByPlaceholderText('start typing to search');
    const changeEvent = createEvent.change(searchEl, {target: {value: 'cat'}});

    fireEvent.change(searchEl, changeEvent);

    expect(spy.mock.calls).toEqual([['cat']]);
  });

  it('should pass "phrase" prop into search input value', () => {
    const {getByPlaceholderText} = render(
      <Search phrase="tiger" setPhrase={() => {}} />,
    );
    const searchInput = getByPlaceholderText('start typing to search');

    expect(searchInput.value).toEqual('tiger');
  });
});

describe('withOnChange()', () => {
  it('should return function', () => {
    expect(typeof withOnChange(() => {})).toBe('function');
  });

  it('should always return new function', () => {
    const funcToWrap = () => {};
    const firstFunc = withOnChange(funcToWrap);
    const secondFunc = withOnChange(funcToWrap);

    expect(firstFunc === secondFunc).toBeFalsy();
  });

  it('should return function which called with ChangeEvent as an argument, will use "target.value" property', () => {
    const spy = jest.fn();
    const funcToTest = withOnChange(spy);
    const changeEventMock = {
      target: {
        value: 'mock',
      },
    };

    funcToTest(changeEventMock);

    expect(spy.mock.calls).toEqual([['mock']]);
  });
});
