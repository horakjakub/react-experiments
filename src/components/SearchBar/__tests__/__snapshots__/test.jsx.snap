// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Search /> renders correctly 1`] = `
<div>
  <div
    class="sc-AxjAm bhPPjN"
  >
    <svg
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      stroke-width="0"
      style="color: rgb(153, 153, 153); font-size: 2em; margin: 0px .6em;"
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
      />
    </svg>
    <input
      class="css-4y926w"
      id="searchVideo"
      name="searchVideo"
      placeholder="start typing to search"
      type="text"
      value=""
    />
  </div>
</div>
`;
